src/compiler/program.ts

```diff
@@ -626,13 +627,29 @@ namespace ts {
                 onlyRecordFailures = !directoryProbablyExists(directory, state.host);
             }
         }
-        return forEach(extensions, tryLoad);
 
-        function tryLoad(ext: string): string {
-            if (ext === ".tsx" && state.skipTsx) {
-                return undefined;
+        if (state.skipTsx)
+            extensions = filter(extensions, ext => ext !== "tsx");

```

Comment:
i would rather we checked this tryLoad to avoid the array allocation for every module lookup.